### PR TITLE
[GLUTEN-5341] Support iceberg bucket join for Spark3.5

### PR DESCRIPTION
**Pull request by yma11**
## What changes were proposed in this pull request?

This PR provides bucket joins support for Spark3.5 to fix corresponding UT failures.

## How was this patch tested?

UT

### CODE REVIEW

**Comment by yma11** [comment]
This PR is pending on fix of [5362](https://github.com/apache/incubator-gluten/issues/5362).

**Comment by yma11** [comment]
@SinghAsDev Can you help review this PR? Thanks.

File: gluten-iceberg/src/test/scala/org/apache/gluten/execution/VeloxIcebergSuite.scala [hunk 1]
```diff
@@ -58,8 +58,7 @@ class VeloxIcebergSuite extends WholeStageTransformerSuite {
     }
   }
 
-  // Ignored due to failures, see https://github.com/apache/incubator-gluten/issues/5362
-  ignore("iceberg bucketed join", Some("3.4")) {
+  testWithSpecifiedSparkVersion("iceberg bucketed join", Some("3.4")) {
```
Comment (SinghAsDev) [inline gluten-iceberg/src/test/scala/org/apache/gluten/execution/VeloxIcebergSuite.scala:61]: Should we enable these tests for 3.5 as well?

**Review by SinghAsDev — APPROVED** [review]

File: gluten-iceberg/src/test/scala/org/apache/gluten/execution/VeloxIcebergSuite.scala (hunk 1 above)
Comment (yma11) [reply gluten-iceberg/src/test/scala/org/apache/gluten/execution/VeloxIcebergSuite.scala:61]: Thanks for comments. The `Some("3.4")` means that this UT will be executed for Spark with version >= 3.4.

**Comment by SinghAsDev** [comment]
Great, thanks!
Best Regards,
Ashish

**Comment by yma11** [comment]
Hi @zhouyuan can you help review this PR? Thanks.

**Review by zhouyuan — APPROVED** [review]

**Pull request merged**